UM/Signal.py

```diff
@@ -121,6 +134,14 @@ def getType(self):
     #   function will be called on the next application event loop tick.
     @call_if_enabled(_traceEmit, _isTraceEnabled())
     def emit(self, *args, **kwargs):
+        FlameProfiler.updateProfileConfig()
```

Comment:
I've changed and moved most of the profiling stuff for emit() such that there is no overheat at runtime. Except for the one big `if` statement in the middle. There are ways to eliminate that too but I fear the result will be too obscure code-wise to justify.